torch/csrc/jit/codegen/cuda/ir_utils.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+class Statement;
+
+namespace ir_utils {
+
+template <
+    typename FilterType,
+    template <typename, typename...> typename Container,
+    typename ElementType>
+Container<FilterType*> filterVals(const Container<ElementType*>& container) {
```

Comment:
why does it have to return the same container as the source?

Reply:
Just for simplicity.

Reply:
Wouldn't be simpler (for the non-iterator version) to just return a `std::vector<FilterType*>`? That avoids the whole complication with trying to deduce the container type, and leads to more predictable behavior too (we wouldn't necessarily want to use a std::list or std::deque to return the results just to match the original container)